src/translators/sql_translator.py

```diff
@@ -455,8 +455,175 @@ def is_text(attr_type):
+                return float(attr['value'])
+        except ValueError:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+
+    @staticmethod
+    def _ngsi_datetime_to_db(attr):
+        if SQLTranslator._is_iso_date(attr['value']):
+            return attr['value']
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+    @staticmethod
+    def _ngsi_integer_to_db(attr):
+        try:
+            if isinstance(attr['value'], bool):
+                return None
+            elif isinstance(attr['value'], int):
+                return attr['value']
+            elif attr['value'] is not None:
+                return int(float(attr['value']))
+        except ValueError:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+    @staticmethod
+    def _ngsi_boolean_to_db(attr):
+        if isinstance(attr['value'], str) and attr['value'].lower() == 'true':
+            return True
+        elif isinstance(attr['value'], str) \
+                and attr['value'].lower() == 'false':
+            return False
+        elif isinstance(attr['value'], int) and attr['value'] == 1:
+            return True
+        elif isinstance(attr['value'], int) and attr['value'] == 0:
+            return False
+        elif isinstance(attr['value'], bool):
+            return attr['value']
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+            return None
+
+    @staticmethod
+    def _ngsi_slf_to_db(attr):
+        raise NotImplementedError
+
+    @staticmethod
+    def _ngsi_structured_to_db(attr):
+        raise NotImplementedError
+
+    @staticmethod
+    def _ngsi_array_to_db(attr):
+        raise NotImplementedError
+
+    @staticmethod
+    def _ngsi_text_to_db(attr):
+        if 'value' in attr and attr['value'] is not None:
+            return str(attr['value'])
+        logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value'], attr['type']))
+        return None
+
+    @staticmethod
+    def _ngsi_default_to_db(attr):
+        return attr.get('value', None)
+
+    @staticmethod
+    def _ngsi_ld_datetime_to_db(attr):
+        if SQLTranslator._is_iso_date(attr['value']['@value']):
+            return attr['value']['@value']
+        else:
+            logging.warning('{} cannot be cast to {} replaced with None'.format(attr['value']['@value'], attr['value']['@type']))
```

Comment:
Is this going to work in all cases? what if `attr` has no `@type` attribute in it?